examples/gallery/simple_differential_rotation.py

```diff
@@ -0,0 +1,72 @@
+"""
+============================
+Simple Differential Rotation
+============================
+
+The Sun is known to rotate differentially, meaning that the rotation rate
+near the poles (rotation period of approximately 35 days) is not the same as
+the rotation rate near the equator (rotation period of approximately 25 days).
+This is possible because the Sun is not a solid body. Though it is still poorly
+understood, it is fairly well measured and must be taken into account
+when comparing observations of features on the Sun over time.
+A good review can be found in Beck 1999 Solar Physics 191, 47–70.
+This example explores the effects of differential rotation.
+"""
+
+##############################################################################
+# Start by importing the necessary modules.
+
+from __future__ import print_function, division
+from datetime import timedelta
+
+import numpy as np
+import matplotlib.pyplot as plt
+import astropy.units as u
+
+import sunpy.map
+import sunpy.data.sample
+from sunpy.physics.differential_rotation import diff_rot, rot_hpc
+
+##############################################################################
+# Next lets quickly explore differential rotation by replicating Figure 1
+# in Beck 1999
+latitudes = u.Quantity(np.arange(0, 90, 1), 'deg')
+dt = 1 * u.day
+rotation_rate = [diff_rot(dt, this_lat) / dt for this_lat in latitudes]
+rotation_period = [360 * u.deg / this_rate for this_rate in rotation_rate]
+
+fig = plt.figure()
+plt.plot(np.sin(latitudes), [this_period.value for this_period in rotation_period])
+plt.ylim(38, 24)
+plt.ylabel('Rotation Period [{0}]'.format(rotation_period[0].unit))
+plt.xlabel('Sin(Latitude)')
+plt.title('Solar Differential Rotation Rate')
+
+##############################################################################
+# Next let's show how to this looks like on the Sun.
+# Load in an AIA map:
+aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
+
+##############################################################################
+# Let's define our starting coordinates
+hpc_y = u.Quantity(np.arange(-700, 800, 100), u.arcsec)
+hpc_x = u.Quantity(np.zeros(len(hpc_y)), u.arcsec)
```

Comment:
The code pattern `np.zeros(len(hpc_y))` should already typically instead be `np.zeros_like(hpc_y)`, but in this case, you definitely should use instead:
```python
hpc_x = np.zeros_like(hpc_y)
```
which is much cleaner and works exactly as it should with Quantities.